network/framework/src/application/storage.rs

```diff
@@ -345,6 +380,46 @@ impl PeersAndMetadata {
         Ok(())
     }
 
+    fn broadcast(&self, event: ConnectionNotification) {
+        let mut listeners = self.subscribers.write();
+        let mut to_del = vec![];
+        for i in 0..listeners.len() {
+            let dest = listeners.get_mut(i).unwrap();
+            match dest.try_send(event.clone()) {
+                Ok(_) => {},
+                Err(err) => match err {
+                    TrySendError::Full(_) => {
+                        // meh, drop message, maybe counter?
+                    },
+                    TrySendError::Closed(_) => {
+                        to_del.push(i);
+                    },
+                },
+            }
+        }
+        for evict in to_del.into_iter() {
+            let llast = listeners.len() - 1;
```

Comment:
Why not just use the [swap_vector](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.swap_remove) method?